modules/data-entry/src/main/frontend/src/questionnaire/Form.jsx

```diff
@@ -348,22 +365,59 @@ function Form (props) {
         <Grid item className={classes.formHeader} xs={12}>
           { parentDetails && <Typography variant="overline">
             {parentDetails}
-            <Tooltip title="Change subject">
-              <IconButton className={classes.hierarchyEditButton} size="small" onClick={() => {setSelectorDialogOpen(true)}}>
-                <EditIcon fontSize="small" />
-              </IconButton>
-            </Tooltip>
           </Typography> }
           <Typography variant="h2">
             {title}
-            <DeleteButton
-              entryPath={data ? data["@path"] : formURL}
-              entryName={(data?.subject?.identifier || "Subject") + ": " + (title)}
-              entryType={data?.questionnaire?.title || "Form"}
-              shouldGoBack={true}
-              buttonClass={classes.titleButton}
-              onComplete={removeWindowHandlers}
-            />
+            <div className={classes.actionsMenu}>
+                {isEdit ?
+                  <Tooltip title="Save and close" onClick={onClose}>
+                    <IconButton color="primary">
+                      <DoneIcon />
+                    </IconButton>
+                  </Tooltip>
+                  :
+                  <Tooltip title="Edit">
+                    <IconButton color="secondary" onClick={onEdit}>
+                      <EditIcon />
+                    </IconButton>
+                  </Tooltip>
+                }
+                <Tooltip title="More actions" onClick={(event) => {setActionsMenu(event.currentTarget)}}>
+                  <IconButton>
+                    <MoreIcon fontSize="small" />
+                  </IconButton>
+                </Tooltip>
+                <Popover
+                    open={Boolean(actionsMenu)}
+                    anchorEl={actionsMenu}
+                    onClose={() => {setActionsMenu(null)}}
+                    anchorOrigin={{
+                        vertical: 'bottom',
+                        horizontal: 'right',
+                    }}
+                    transformOrigin={{
+                        vertical: 'top',
+                        horizontal: 'right',
+                    }}
+                >
+                  <List>
+                    <ListItem className={classes.actionsMenuItem}>
+                      <Button onClick={() => {setSelectorDialogOpen(true)}}>
```

Comment:
I haven't tested, but this may "close" the menu when clicking on "Change subject".
```suggestion
                      <Button onClick={() => {setSelectorDialogOpen(true); setActionsMenu(null)}}>
```